### PR TITLE
Updates for gcloud 0.9.54

### PR DESCRIPTION
**Pull request by j3ffml**
Updates gke provider and e2e driver for breaking changes in gcloud 0.9.54.

cc @zmerlynn

### CODE REVIEW

File: cluster/gke/util.sh [hunk 1]
```diff
@@ -91,6 +91,7 @@ function verify-prereqs() {
     gcloud_prompt="-q"
   fi
   gcloud ${gcloud_prompt:-} components update preview || true
```
Comment (roberthbailey) [inline cluster/gke/util.sh]: Remove the update preview line?

File: test/e2e/util.go [hunk 2]
```diff
@@ -41,17 +41,18 @@ const (
...
-	gceConfig  GCEConfig
+type TestContextType struct {
+	KubeConfig  string
+	KubeContext string
+	AuthConfig  string
+	CertDir     string
+	Host        string
+	RepoRoot    string
+	Provider    string
+	GceConfig   GCEConfig
```
Comment (roberthbailey) [inline test/e2e/util.go]: should be named GCEConfig, as acronyms are consistent case in Go.

File: cluster/gke/util.sh (hunk 1 above)
Comment (jlowdermilk) [inline cluster/gke/util.sh]: We still need the preview component for instance-groups and managed-instance-groups, which were not moved to alpha.

File: test/e2e/util.go (hunk 2 above)
Comment (jlowdermilk) [inline test/e2e/util.go]: Done.

**Comment by roberthbailey** [comment]
LGTM. Assigning to @zmerlynn to merge once the Jenkins box is updated to the next release of gcloud.

**Comment by jlowdermilk** [comment]
Gcloud 0.9.54 is now live. e2e test in progress.

**Comment by zmerlynn** [comment]
@jlowdermilk and I talked about this. The e2es are passing as well in GKE on the new gcloud as the old. I'm merging this slightly outside the merge window because it's when I'm available to do the Jenkins maintenance and watch it for a bit longer (and it's a quiet period otherwise).

**Pull request merged**